src/stats/targets.ts

```diff
@@ -0,0 +1,71 @@
+import type { FrameEntryType, FramesType, GameStartType } from "../types";
+import { Frames, GameMode } from "../types";
+import type { TargetBreakType } from "./common";
+import type { StatComputer } from "./stats";
+
+// The Target item's in-game ID
+const TARGET_ITEM_TYPE_ID = 209;
+
+export class TargetBreakComputer implements StatComputer<TargetBreakType[]> {
+  private targetBreaks = new Array<TargetBreakType>();
+  private isTargetTestGame = false;
+
+  public setup(settings: GameStartType): void {
+    // Reset the state
+    this.targetBreaks = [];
+    this.isTargetTestGame = settings.gameMode === GameMode.TARGET_TEST;
+  }
+
+  public processFrame(frame: FrameEntryType, allFrames: FramesType): void {
+    if (!this.isTargetTestGame) {
+      return;
+    }
+
+    handleTargetBreak(allFrames, frame, this.targetBreaks);
+  }
+
+  public fetch(): TargetBreakType[] {
+    return this.targetBreaks;
+  }
+}
+
+function handleTargetBreak(frames: FramesType, frame: FrameEntryType, targetBreaks: TargetBreakType[]): boolean {
```

Comment:
does this need to return a boolean? it isn't being used anymore in the one place it gets called